packages/flutter/test/material/text_field_test.dart

```diff
@@ -4632,9 +4630,54 @@ void main() {
     );
 
     await gesture.up();
-    await tester.pumpAndSettle();
+    await tester.pump(const Duration(milliseconds: 500));
```

Comment:
What's the rationale for these changes? What happens in 500ms that doesn't happen upon `pumpAndSettle()`?

Reply:
If it takes 600ms, it's aesthetically incorrect and a bug.

Reply:
Also avoiding pumpAndSettle in general https://docs.flutter.io/flutter/flutter_test/WidgetTester/pumpAndSettle.html

Reply:
I don't think it makes sense to avoid replace pumpAndSettle() with arbitrary time limits. Where does 500ms come from?

Reply:
We definitely need to figure out a guideline for pump v pumpAndSettle because it seems there are a lot of different opinions on what is best.

Reply:
Actually toolbars aren't animated. Replaced all with just a one frame pump. 

I strongly advocate making an effort to bring the usages of pumpAndSettle to 0 in our tests. Animations that doesn't take exactly the amount of time as expected is a bug. UI events that takes more post frame callbacks than exactly the number of expected is a bug.

Reply:
I don't think that's a good idea. If you don't have a concrete rationale for a particular time limit then you've added specificity to a test that doesn't belong and will only serve to confuse developers in the future. Many of the time limits in the existing (including this one apparently) are not based on any documented rationale, they're effectively arbitrary.

Reply:
I think tests should test production behavior. If a test doesn't have concrete rationale for the time duration of a UI behavior or if it's arbitrary, it should imply that the production behavior has no rationale for being bound by any time duration or is arbitrary. 

That should only happen when the production behavior is intentionally arbitrary (in which case using pumpAndSettle makes sense). If the production value is not arbitrary, the test should not be arbitrary. In this case, it's a single frame.

Reply:
Sure, but you can't just make up "production values", they need to be based on real expectations about behavior. And if you're going to include such values then you need to explain what they're based on. Otherwise you're just adding noise to the test implementation.

Reply:
If you meant all pumps with a specific duration should have an accompanying code comment, I agree.

Reply:
i.e. the way I had it originally is incorrect

Reply:
It can take a considerable amount of analysis to decide exactly how much time and/or how many frames a change needs before the state of the UI can be verified.  When a test author is willing to do that and to explain it in a comment, then I agree that explicit pumps and durations are preferred. When there isn't time to do any of that, I think using pumpAndSettle() is preferable, since it indicates an opportunity to tighten up the test.